### DISCUSSION TITLE
is it possible to pass ctx to a templ.handler(component)?

### DISCUSSION THREAD

**Discussion by JuLi0n21** [a-h/templ#674] [Q&A] [answered]
i want to use adaptive theming for resuing components on diffrent sites with changing styles, i wanna avoid propdrilling because i have a 3-5 layers of nested components 

```
themes := types.Theme{
	Dark:         "backdrop--dark",
	Medium_dark:  "backdrop--medium-dark",
	Medium:       "backdrop--medium",
	Medium_light: "backdrop--medium-light",
	Light:        "backdrop--light",
}
	
ctx := context.WithValue(context.Background(), "theme", themes)

component := views.ScoreSite(player, scores) <--pass ctx to component

handler := adaptor.HTTPHandler(templ.Handler(component)) 
```

**Comment by a-h** [comment, marked as answer]
Yes, there's an explanation of how to use context to avoid prop drilling in the docs at https://templ.guide/syntax-and-usage/context/

And how to create middleware to pass the context down at https://templ.guide/syntax-and-usage/context/#using-context-with-http-middleware

**Reply by JuLi0n21** [reply]
thanks worked
